crates/iota-node/src/lib.rs

```diff
@@ -1464,15 +1482,16 @@ impl IotaNode {
 
         server_builder = server_builder.add_service(ValidatorServer::new(validator_service));
 
-        let server = server_builder
+        let mut server = server_builder
             .bind(config.network_address())
             .await
             .map_err(|err| anyhow!(err.to_string()))?;
+        let cancel_handle = server.take_cancel_handle().unwrap();
```

Comment:
I suppose unwrapping is okay here since this only gets run once at node startup ? Also should we use expect instead of unwrap to add error context ?